Makefile

```diff
@@ -38,10 +47,10 @@ ifneq (,$(findstring clang,$(CC_VERSION)))
 endif
 
 ifdef BCACHEFS_DEBUG
-	CFLAGS+=-Werror
-	CFLAGS+=-DCONFIG_BCACHEFS_DEBUG=y
+	CFLAGS+=-Werror						\
+		-DCONFIG_BCACHEFS_DEBUG=y			\
+		-DCONFIG_VALGRIND=y
```

Comment:
with your make debug target we can just get rid of BCACHEFS_DEBUG

Reply:
I'm not sure this is what you want (or not yet): A developer may very well like the fact he could set BCACHEFS_DEBUG=1 in .profile and always create a debug build (which is what most people are doing right now, no?).  Unless you're sure nobody cares, removing it may need to be postponed